proc/proc_windows.go

```diff
@@ -507,39 +536,65 @@ func (dbp *Process) wait(pid, options int) (int, *sys.WaitStatus, error) {
 }
 
 func (dbp *Process) setCurrentBreakpoints(trapthread *Thread) error {
-	// TODO: In theory, we should also be setting the breakpoints on other
-	// threads that happen to have hit this BP. But doing so leads to periodic
-	// failures in the TestBreakpointsCounts test with hit counts being too high,
-	// which can be traced back to occurrences of multiple threads hitting a BP
-	// at the same time.
+	err := trapthread.SetCurrentBreakpoint()
+	if err != nil {
+		return err
+	}
 
-	// My guess is that Windows will correctly trigger multiple DEBUG_EVENT's
-	// in this case, one for each thread, so we should only handle the BP hit
-	// on the thread that the debugger was evented on.
+	for _, thread := range dbp.Threads {
+		thread.running = false
+		_, err := _SuspendThread(thread.os.hThread)
+		if err != nil {
+			return err
+		}
+	}
 
-	return trapthread.SetCurrentBreakpoint()
+	for {
+		var err error
+		var tid int
+		dbp.execPtraceFunc(func() {
+			err = _ContinueDebugEvent(uint32(dbp.Pid), uint32(dbp.os.breakThread), _DBG_CONTINUE)
```

Comment:
If we suspend all threads (which we did above this line) `ContinueDebugEvent` doesn't resume execution, it just acknowledges that we have processed the event we received. I couldn't figure out any other way of doing this that would work with the windows API. If we don't do this later on on the Continue procedure while we are singlestepping CurrentThread over its breakpoint instruction we receive events for other threads and we are not equipped to dealing with them there.